src/App.tsx

```diff
@@ -1,13 +1,41 @@
-import React from 'react';
+import React, {useState} from 'react';
 import { Grid } from '@mui/material';
 import { NetworkGraphComponent } from './components/NetworkGraph/NetworkGraphComponent';
 import { HeaderComponent } from './components/Header/HeaderComponent';
 import { SidebarComponent } from './components/Details/SidebarComponent';
 
-export const App = () => (
-  <div>
-    <HeaderComponent />
-    <NetworkGraphComponent />
-    <SidebarComponent />
-  </div>
-);
+export const App = () => {
+
+    const [panelWidth, setPanelWidth] = useState(350);
+
+    const handlePanelDragSelect = (event: React.MouseEvent<HTMLDivElement, MouseEvent>) => {
+        document.addEventListener('mousemove', handlePanelDrag)
+        document.addEventListener('mouseup', handlePanelDragUnselect)
+        document.body.style.userSelect = 'none'
+    }
+
+    const handlePanelDrag = (event: MouseEvent) => {
+        let offsetRight =
+            document.body.offsetWidth - (event.clientX - document.body.offsetLeft) + 20;
+        let minWidth = 50;
+        let maxWidth = 600;
+        if (offsetRight > minWidth && offsetRight < maxWidth) {
+            setPanelWidth(offsetRight);
+        }
+    }
+
+    const handlePanelDragUnselect = () => {
+        console.log('Mouse Up')
```

Comment:
Try to not leave in debugging console.log statements like this. Or at least use console.debug().